prusa3d/Prusa-Firmware
# cmake build improvements

In this PR we continue the work to improve the cmake build system:

- Allow to restrict the build to selected variants
- Move Tests into tests/ and prune obsolete ones
- Update Catch2 and related infrastructure
- Make the build finally reproducible
- Tweak AVR target flags to match Arduino

This makes the EN-only build complete.
There are still some things to be completed:

- [ ] verify multi-language builds for changes
- [ ] set the correct source epoch based on the source commit date
